components/components.go

```diff
@@ -2,144 +2,133 @@
 package components
 
 import (
-	"context"
 	"fmt"
 	"sync"
-	"time"
 
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
-	"github.com/leptonai/gpud/pkg/common"
 	"github.com/leptonai/gpud/pkg/errdefs"
 )
 
-// Component represents an individual component of the system.
-//
-// Each component check is independent of each other.
-// But the underlying implementation may share the same data sources
-// in order to minimize the querying overhead (e.g., nvidia-smi calls).
-//
-// Each component implements its own output format inside the State struct.
-// And recommended to have a consistent name for its HTTP handler.
-// And recommended to define const keys for the State extra information field.
-type Component interface {
-	// Defines the component name,
-	// and used for the HTTP handler registration path.
-	// Must be globally unique.
-	Name() string
-
-	// Start called upon server start.
-	// Implements component-specific poller start logic.
-	Start() error
-
-	// Returns the current states of the component.
-	States(ctx context.Context) ([]State, error)
-
-	// Returns all the events from "since".
-	Events(ctx context.Context, since time.Time) ([]Event, error)
-
-	// Called upon server close.
-	// Implements copmonent-specific poller cleanup logic.
-	Close() error
-}
-
-// HealthSettable is an optional interface that can be implemented by components
-// to allow setting the health state.
-type HealthSettable interface {
-	// SetHealthy sets the health state to healthy.
-	SetHealthy() error
-}
-
-type State struct {
-	Name      string            `json:"name,omitempty"`
-	Healthy   bool              `json:"healthy,omitempty"`
-	Health    string            `json:"health,omitempty"`     // Healthy, Degraded, Unhealthy
-	Reason    string            `json:"reason,omitempty"`     // a detailed and processed reason on why the component is not healthy
-	Error     string            `json:"error,omitempty"`      // the unprocessed error returned from the component
-	ExtraInfo map[string]string `json:"extra_info,omitempty"` // any extra information the component may want to expose
+var (
+	defaultSetMu sync.RWMutex
+	defaultSet   = make(map[string]Component)
+)
 
-	SuggestedActions *common.SuggestedActions `json:"suggested_actions,omitempty"`
+// GetAllComponents returns all the components in the default set.
+func GetAllComponents() map[string]Component {
+	defaultSetMu.RLock()
+	defer defaultSetMu.RUnlock()
+	return defaultSet
```

Comment:
It's not better to directly return the map since it may bring potential races. shall we return a cloned map?

Reply:
addressed